### DISCUSSION TITLE
Using clink via ssh without outrun

### DISCUSSION THREAD

**Discussion by pmeheut** [chrisant996/clink#381] [Q&A] [answered]
Hi,
I have to connect via ssh to an user PC and I would like to run clink when I do so.
But I do not want to use outrun because I want him to keep his "normal" cmd.exe.

I've tried to ssh with `cmd.exe /k clink_x64.exe inject` and `cmd.exe /k clink.bat` but it does not work: the command runs but does nothing.
Any idea?

**Comment by chrisant996** [comment]
@pmeheut Can you clarify a few things, please:

- Can you share the clink.log file?
- Does "outrun" refer to https://github.com/Overv/outrun?
- Are you saying that Clink works over ssh when using outrun, but you don't want to use outrun?

I wouldn't have expected for Clink to work over ssh.  It has to use a virtual terminal input keyboard driver that interacts directly with the console input subsystem, and I don't see how that could work over ssh.  But if you're saying that there are situations where Clink has its full capabilities when running over ssh, then can you share more information about that?

I haven't used ssh, and I don't have access to an ssh server.  My understanding is it essentially redirects stdin, stdout, and stderr across encrypted sockets.  That won't have access to the console subsystem, and so input will be limited to letters and symbols and a subset of Ctrl codes.  I don't know whether Lua scripts and prompt filtering would be expected to work.

**Comment by pmeheut** [comment]
Sorry: outrun is the spell-checker at work. I meant autorun.

I understand your explanation and clink.log says:

05c8 start_logger              160 ---- 2022/12/06 06:16:22.592 -------------------------------------------------
05c8 start_logger              166 Host process is 'cmd.exe' (pid 1480)
05c8 start_logger              170 DLL path is 'C:\Users\<username>\Desktop\clink'
05c8 start_logger              185 Windows version 10.0.17763 (x64)
05c8 start_logger              188 Clink version 1.4.3.67deea (x64)
05c8 host_cmd::validate        306 Host is not interactive; cancelling inject.

**Comment by pmeheut** [comment, marked as answer]
BTW, reading the clink.log gave me an idea and I found the solution. It has nothing to do with clink. I just needed to pass the '-t' option to ssh. This is unnecessary usually but here, with a command to run, it forces it to allocate a console subsystem.
So everything works.
Thanks.

**Reply by chrisant996** [reply]
Glad you figured it out, and thanks for sharing the solution.
